dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/handling/DataHandler.java

```diff
@@ -0,0 +1,1161 @@
+
+            Map<String, Double> dataMap = metric != EXPECTED_REPORTS
+                ? getAggregatedCompletenessValueMap( params )
+                : new HashMap<>();
+
+            Integer periodIndex = params.getPeriodDimensionIndex();
+            Integer dataSetIndex = DX_INDEX;
+            Map<String, PeriodType> dsPtMap = params.getDataSetPeriodTypeMap();
+            PeriodType filterPeriodType = params.getFilterPeriodType();
+
+            int timeUnits = getTimeUnits( params );
+
+            for ( Map.Entry<String, Double> entry : targetMap.entrySet() )
+            {
+                List<String> dataRow = newArrayList( entry.getKey().split( DIMENSION_SEP ) );
+
+                Double target = entry.getValue();
+                Double actual = firstNonNull( dataMap.get( entry.getKey() ), 0d );
+
+                if ( target != null )
+                {
+                    // ---------------------------------------------------------
+                    // Multiply target value by number of periods in time span
+                    // ---------------------------------------------------------
+
+                    PeriodType queryPt = filterPeriodType != null ? filterPeriodType
+                        : getPeriodTypeFromIsoString( dataRow.get( periodIndex ) );
+                    PeriodType dataSetPt = dsPtMap.get( dataRow.get( dataSetIndex ) );
+
+                    target = getCalculatedTarget( periodIndex, timeUnits, dataRow, target, queryPt, dataSetPt );
+
+                    addReportRateToGrid( params, grid, metric, dataRow, target, actual );
+                }
+            }
+        }
+    }
+
+    private int getTimeUnits( DataQueryParams params )
+    {
+        return params.hasFilter( PERIOD_DIM_ID ) ? params.getFilterPeriods().size() : 1;
+    }
+
+    /**
+     * Calculate reporting rate and replace data set with rate and add the rate to
+     * the Grid.
+     * 
+     * @param params
+     * @param grid
+     * @param metric
+     * @param dataRow
+     * @param target
+     * @param actual
+     */
+    private void addReportRateToGrid( DataQueryParams params, Grid grid, ReportingRateMetric metric,
+        List<String> dataRow, Double target, Double actual )
+    {
+        Double value = getReportRate( metric, target, actual );
+
+        String reportingRate = getDimensionItem( dataRow.get( DX_INDEX ), metric );
+        dataRow.set( DX_INDEX, reportingRate );
+
+        grid.addRow()
+            .addValues( dataRow.toArray() )
+            .addValue( params.isSkipRounding() ? value : getRounded( value ) );
+
+        if ( params.isIncludeNumDen() )
+        {
+            grid.addValue( actual )
+                .addValue( target )
+                .addValue( PERCENT )
+                .addNullValues( 2 );
+        }
+    }
+
+    /**
+     * Use number of days for daily data sets as target, as query periods might
+     * often span/contain different numbers of days.
+     * 
+     * @param periodIndex
```

Comment:
Add missing javadoc for the method arguments or remove `@param`

Reply:
Done